### ISSUE TITLE
Add TypeScript definitions

### ISSUE THREAD

**Issue by mrdbarker** [pawelgrzybek/siema#20]
I'd like to add TypeScript definitions for Siema, is this something you would be happy to accept as a PR? I have this already written and it would help users that are importing Siema in TypeScript files / users of VS Code.

**Comment by pawelgrzybek** [comment]
Woho! Mike, thats amazing! I doubt that I should merge typeScript version with master branch. I just want to make contribution / readability easier. There is lots of devs who are not familiar with TS (including myself).

As familiarising myself with typescript is on my roadmap and I'm just waiting for a spare moment in life to do it, I would love to see TypeScript'ed Siema from you :) I really appreciate your amazing idea and contribution!

**Comment by mrdbarker** [comment]
@pawelgrzybek Just to clarify, I wouldn't be changing the current Javascript only nature of Siema. TypeScript definitions can be separate from the codebase itself in their own `d.ts` file and then you can set the `"typings"` property inside the `package.json` to tell TypeScript where to find the types.

A good example of this is Vue.js. They author in Javascript and then add TypeScript types after the fact, as seen here:

https://github.com/vuejs/vue/tree/5c34b1ba731ffc53aa10c95bbf5d943f0415372f/types
https://github.com/vuejs/vue/blob/5c34b1ba731ffc53aa10c95bbf5d943f0415372f/package.json#L6

Happy to fork and submit a PR so you can see what I mean!

**Comment by pawelgrzybek** [comment]
Lets postpone this one till weekend my friend, I'll review properly Vue (love vue by the way) examples and let you know. Seems to be amazing idea for me. I just need to familiarise myself with concept first :)

**Comment by mrdbarker** [comment]
@pawelgrzybek Sounds like a good plan! I'll include my work on it so far to help you out :)

```
declare module "siema" {
	interface ISiemaConfig {
		selector?: string | HTMLElement;
		duration?: number;
		easing?: string;
		perPage?: number;
		startIndex?: number;
		draggable?: boolean;
		threshold?: number;
		loop?: boolean;
	}

	interface ISiemaInstance {
		config: ISiemaConfig;
		currentSlide: number;
		drag: {
			end: number;
			start: number;
		}

		innerElements: Array<HTMLElement>;
		perPage: number;
		pointerDown: boolean;
		selector: string | HTMLElement;
		selectorWidth: number;
		sliderFrame: HTMLElement;
	}

	interface ISiema {
		new(options?: ISiemaConfig): ISiemaInstance & ISiema;

		clearDrag(this: ISiema): void;
		goTo(this: ISiema, index: number): void;
		init(this: ISiema): void;
		mousedownHandler(this: ISiema, event: Event): void;
		mouseleaveHandler(this: ISiema, event: Event): void;
		mousemoveHandler(this: ISiema, event: Event): void;
		mouseupHandler(this: ISiema, event: Event): void;
		next(this: ISiema): void;
		resize(this: ISiema): void;
		resolveSlidesNumber(this: ISiema): void;
		slideToCurrent(this: ISiema): void;
		touchendHandler(this: ISiema, event: Event): void;
		touchmoveHandler(this: ISiema, event: Event): void;
		touchstartHandler(this: ISiema, event: Event): void;
		updateAfterDrag(this: ISiema): void;
	}

	const Siema: ISiema;

	export = Siema;
}
```

**Comment by pawelgrzybek** [comment]
WOW! Thanks a ton again! I promise to learn up a bit of TS and then I will add it Mike. Thanks a ton again! Hopefully you will build a cool things with Siema! Have a great day!

**Comment by domoritz** [comment]
@pawelgrzybek Thanks for the awesome library. I was wondering whether you still plan to include the typings in the repo?

**Comment by pawelgrzybek** [comment]
It is coming with 1.5 :) Thanks for your kind words!

**Comment by mrdbarker** [comment]
Great stuff, Pawel :)

**Comment by AdamLyon** [comment]
The typings didn't get released with 1.5, is this still planned?
